### ISSUE TITLE
MissingResourceException when running on iOS (iPhone simulator)

### ISSUE THREAD

**Issue by MathiasDL** [tamimattafi/krop#5]
When trying to run this on iOS, I encounter following exception when trying to run the cropper: 
**MissingResourceException: Missing resource with path: .../AppName.app/compose-resources/drawable/restore.xml**

<details><summary><h3>Stacktrace</h3></summary>

```
Uncaught Kotlin exception: org.jetbrains.compose.resources.MissingResourceException: Missing resource with path: /Users/mdl/Library/Developer/CoreSimulator/Devices/device-uuid/data/Containers/Bundle/Application/app-uuid/AppName.app/compose-resources/drawable/restore.xml
    at 0   kfun:kotlin.Throwable#<init>(kotlin.String?){} + 119 (/opt/buildAgent/work/ed783494cd2364bc/kotlin/kotlin-native/runtime/src/main/kotlin/kotlin/Throwable.kt:28:44)
    at 1   kfun:kotlin.Exception#<init>(kotlin.String?){} + 115 (/opt/buildAgent/work/ed783494cd2364bc/kotlin/kotlin-native/runtime/src/main/kotlin/kotlin/Exceptions.kt:23:51)
    at 2   kfun:org.jetbrains.compose.resources.MissingResourceException#<init>(kotlin.String){} + 179 (/opt/buildAgent/work/3eb902016bd3838/components/resources/library/src/commonMain/kotlin/org/jetbrains/compose/resources/ResourceReader.kt:5:48)
    at 3   kfun:org.jetbrains.compose.resources.object-2.readData#internal + 515 (/opt/buildAgent/work/3eb902016bd3838/components/resources/library/src/iosMain/kotlin/org/jetbrains/compose/resources/ResourceReader.ios.kt:28:77)
    at 4   kfun:org.jetbrains.compose.resources.object-2.read#internal + 395 (/opt/buildAgent/work/3eb902016bd3838/components/resources/library/src/iosMain/kotlin/org/jetbrains/compose/resources/ResourceReader.ios.kt:10:20)
    at 5   kfun:org.jetbrains.compose.resources.ResourceReader#read#suspend(kotlin.String;kotlin.coroutines.Continuation<kotlin.ByteArray>){}kotlin.Any-trampoline + 115 (/opt/buildAgent/work/3eb902016bd3838/components/resources/library/src/commonMain/kotlin/org/jetbrains/compose/resources/ResourceReader.kt:26:13)
    at 6   kfun:org.jetbrains.compose.resources.$loadImage$lambda$15COROUTINE$3.invokeSuspend#internal + 371 (/opt/buildAgent/work/3eb902016bd3838/components/resources/library/src/commonMain/kotlin/org/jetbrains/compose/resources/ImageResources.kt:150:68)
    at 7   kfun:org.jetbrains.compose.resources.loadImage$lambda$15#internal + 395 (/opt/buildAgent/work/3eb902016bd3838/components/resources/library/src/commonMain/kotlin/org/jetbrains/compose/resources/ImageResources.kt:150:44)
    at 8   kfun:org.jetbrains.compose.resources.$loadImage$lambda$15$FUNCTION_REFERENCE$12.invoke#internal + 115 (/opt/buildAgent/work/3eb902016bd3838/components/resources/library/src/commonMain/kotlin/org/jetbrains/compose/resources/ImageResources.kt:150:28)
    at 9   kfun:kotlin.coroutines.SuspendFunction0#invoke#suspend(kotlin.coroutines.Continuation<1:0>){}kotlin.Any?-trampoline + 107 (/Users/teamcity/.gradle/daemon/8.8/[K][Suspend]Functions:1:1)
    at 10  kfun:org.jetbrains.compose.resources.AsyncCache.getOrLoad$lambda$1$lambda$0#internal + 115 (/opt/buildAgent/work/3eb902016bd3838/components/resources/library/src/commonMain/kotlin/org/jetbrains/compose/resources/AsyncCache.kt:19:63)
    at 11  kfun:org.jetbrains.compose.resources.AsyncCache.$getOrLoad$lambda$1$lambda$0$FUNCTION_REFERENCE$1.invoke#internal + 135 (/opt/buildAgent/work/3eb902016bd3838/components/resources/library/src/commonMain/kotlin/org/jetbrains/compose/resources/AsyncCache.kt:19:26)
    at 12  kfun:kotlin.Function2#invoke(1:0;1:1){}1:2-trampoline + 115 (/Users/teamcity/.gradle/daemon/8.8/[K][Suspend]Functions:1:1)
    at 13  kfun:kotlin.coroutines.intrinsics.object-4.invokeSuspend#internal + 739 (/opt/buildAgent/work/ed783494cd2364bc/kotlin/kotlin-native/runtime/src/main/kotlin/kotlin/coroutines/intrinsics/IntrinsicsNative.kt:254:35)
    at 14  kfun:kotlin.coroutines.native.internal.BaseContinuationImpl#invokeSuspend(kotlin.Result<kotlin.Any?>){}kotlin.Any?-trampoline + 71 (/opt/buildAgent/work/ed783494cd2364bc/kotlin/kotlin-native/runtime/src/main/kotlin/kotlin/coroutines/ContinuationImpl.kt:50:24)
    at 15  kfun:kotlin.coroutines.native.internal.BaseContinuationImpl#resumeWith(kotlin.Result<kotlin.Any?>){} + 647 (/opt/buildAgent/work/ed783494cd2364bc/kotlin/kotlin-native/runtime/src/main/kotlin/kotlin/coroutines/ContinuationImpl.kt:30:39)
    at 16  kfun:kotlin.coroutines.Continuation#resumeWith(kotlin.Result<1:0>){}-trampoline + 99 (/opt/buildAgent/work/ed783494cd2364bc/kotlin/libraries/stdlib/src/kotlin/coroutines/Continuation.kt:26:12)
    at 17  kfun:kotlinx.coroutines.DispatchedTask#run(){} + 1895 (/opt/buildAgent/work/44ec6e850d5c63f0/kotlinx-coroutines-core/common/src/internal/DispatchedTask.kt:101:71)
    at 18  kfun:kotlinx.coroutines.Runnable#run(){}-trampoline + 91 (/opt/buildAgent/work/44ec6e850d5c63f0/kotlinx-coroutines-core/native/src/Runnable.kt:10:5)
    at 19  kfun:kotlinx.coroutines.EventLoopImplBase#processNextEvent(){}kotlin.Long + 291 (/opt/buildAgent/work/44ec6e850d5c63f0/kotlinx-coroutines-core/common/src/EventLoop.common.kt:263:44)
    at 20  kfun:kotlinx.coroutines.EventLoop#processNextEvent(){}kotlin.Long-trampoline + 55 (/opt/buildAgent/work/44ec6e850d5c63f0/kotlinx-coroutines-core/common/src/EventLoop.common.kt:49:5)
    at 21  kfun:kotlinx.coroutines.BlockingCoroutine.joinBlocking#internal + 435 (/opt/buildAgent/work/44ec6e850d5c63f0/kotlinx-coroutines-core/native/src/Builders.kt:129:43)
    at 22  kfun:kotlinx.coroutines#runBlocking(kotlin.coroutines.CoroutineContext;kotlin.coroutines.SuspendFunction1<kotlinx.coroutines.CoroutineScope,0:0>){0§<kotlin.Any?>}0:0 + 1455 (/opt/buildAgent/work/44ec6e850d5c63f0/kotlinx-coroutines-core/native/src/Builders.kt:68:26)
    at 23  kfun:kotlinx.coroutines#runBlocking$default(kotlin.coroutines.CoroutineContext?;kotlin.coroutines.SuspendFunction1<kotlinx.coroutines.CoroutineScope,0:0>;kotlin.Int){0§<kotlin.Any?>}0:0 + 255 (/opt/buildAgent/work/44ec6e850d5c63f0/kotlinx-coroutines-core/native/src/Builders.kt:45:1)
    at 24  kfun:org.jetbrains.compose.resources#rememberResourceState(kotlin.Any;kotlin.Any;kotlin.Any;kotlin.Function0<0:0>;kotlin.coroutines.SuspendFunction1<org.jetbrains.compose.resources.ResourceEnvironment,0:0>;androidx.compose.runtime.Composer?;kotlin.Int){0§<kotlin.Any?>}androidx.compose.runtime.State<0:0> + 1623 (/opt/buildAgent/work/3eb902016bd3838/components/resources/library/src/blockingMain/kotlin/org/jetbrains/compose/resources/ResourceState.blocking.kt:46:13)
    at 25  kfun:org.jetbrains.compose.resources#vectorResource(org.jetbrains.compose.resources.DrawableResource;androidx.compose.runtime.Composer?;kotlin.Int){}androidx.compose.ui.graphics.vector.ImageVector + 1387 (/opt/buildAgent/work/3eb902016bd3838/components/resources/library/src/commonMain/kotlin/org/jetbrains/compose/resources/ImageResources.kt:83:24)
    at 26  kfun:org.jetbrains.compose.resources#painterResource(org.jetbrains.compose.resources.DrawableResource;androidx.compose.runtime.Composer?;kotlin.Int){}androidx.compose.ui.graphics.painter.Painter + 2059 (/opt/buildAgent/work/3eb902016bd3838/components/resources/library/src/commonMain/kotlin/org/jetbrains/compose/resources/ImageResources.kt:40:38)
    at 27  kfun:com.attafitamim.krop.ui.ComposableSingletons$ImageCropperDialogKt.<init>$lambda$4#internal + 507 (/Users/runner/work/krop/krop/library/ui/src/commonMain/kotlin/com/attafitamim/krop/ui/ImageCropperDialog.kt:100:22)
    at 28  kfun:com.attafitamim.krop.ui.ComposableSingletons$ImageCropperDialogKt.$<init>$lambda$4$FUNCTION_REFERENCE$10.invoke#internal + 95 (/Users/runner/work/krop/krop/library/ui/src/commonMain/kotlin/com/attafitamim/krop/ui/ImageCropperDialog.kt:99:53)
    at 29  kfun:com.attafitamim.krop.ui.ComposableSingletons$ImageCropperDialogKt.$<init>$lambda$4$FUNCTION_REFERENCE$10.$<bridge-DNNNU>invoke(androidx.compose.runtime.Composer;kotlin.Int){}#internal + 159 (/Users/runner/work/krop/krop/library/ui/src/commonMain/kotlin/com/attafitamim/krop/ui/ImageCropperDialog.kt:99:53)
...
```

</details>

The stacktrace and error message show that the issue happens in the `ImageCropperDialog.kt` file:

```kotlin
IconButton(onClick = { state.reset() }) {
    Icon(painterResource(Res.drawable.restore), null) // <--
}
```

It seems that the resources are not included in the `.app` file. The resources from my app are in there, and work correctly.

**Comment by tamimattafi** [comment]
@MathiasDL It seems like an issue with compose resources, what's the version of Kotlin in your project? 
Multi-module support was added in Kotlin 2.0, but I'm not sure if the library has to be 2.0 as well, if so, I would be glad to upgrade.

**Comment by MathiasDL** [comment]
I'm indeed on Kotlin version `2.0.20`

**Comment by tamimattafi** [comment]
@MathiasDL Sure, I will work on it as soon as possible

**Comment by tamimattafi** [comment]
@MathiasDL Could you please try this version from PR #7 in your app? 
I tested the sample app on Android and iOS Simulators and it works well.

**Comment by MathiasDL** [comment]
@tamimattafi The sample app for Android and iOS also works on my machine.

For testing with my CMP app, do you know how I can use the library (branch) in my project? I tried [generating an .aar](https://stackoverflow.com/questions/24309950/create-aar-file-in-android-studio), but importing the classes does not work, and it does not include the resources. I also tried this approach:

settings.gradle.kts:
```kotlin
include(":krop")
project(":krop").projectDir = File(settingsDir, "../relativePathTo/krop")
```

build.gradle.kts:
```kotlin
dependencies {
    implementation(project(":krop")) // or api(":krop")
}
```

But then the gradle files get confused: I get an `Unresolved reference: library` about the plugin line: `alias(libs.plugins.android.library).apply(false)`, since it can't find that in **my** libs.versions.toml

**Comment by tamimattafi** [comment]
@MathiasDL You can just publish the library to `mavenLocal`, then implement it as usual.
1. Checkout the branch from PR #7 `config/kotlin_2.0`
2. Inside `PublishConventions.kt` change the version to something like `"0.1.1-local01"`
3. Inside the same file comment out `// signAllPublications()`
4. From the Gradle panel (or in the terminal), run `publishToMavenLocal` in the root folder
5. In your project, change the version to `"0.1.1-local01"` and sync
6. Run your project

**Comment by MathiasDL** [comment]
Thanks for the clear instructions! I just had to add `mavenLocal()` to my repositories, and it worked immediately.

Both the android and iOS (simulator) version work correctly now 🥳

**Comment by tamimattafi** [comment]
@MathiasDL This issue will be fixed in [Release 0.1.1](https://github.com/tamimattafi/krop/releases/tag/0.1.1)
Please check in your project after the library becomes available on `mavenCentral`, and if everything is fine, close the issue. (You can check the progress of this [action](https://github.com/tamimattafi/krop/actions/runs/10935341372))
